gammapy/utils/fitting/scipy.py

```diff
@@ -27,6 +27,82 @@ def optimize_scipy(parameters, function, **kwargs):
     return factors, info, optimizer
 
 
+class TSDifference:
+    """Likelihood wrapper to compute TS differences"""
+    def __init__(self, function, parameters, parameter, ts_diff):
+        self.loglike_ref = function(parameters)
+        self.parameters = parameters
+        self.function = function
```

Comment:
I would call it `ts_function` and `ts_ref`. Since most of the time this is 2*loglike

Reply:
Note that the same function could be used to compute 2D confidence regions.
I would use a more specific name for the tested parameter(s).

Reply:
Concerning the naming of the fit statistics I've made a reminder issue #2149.